recipe/meta.yaml

```
@@ -1,14 +1,13 @@
-{% set name = "dask-ml" %}
-{% set version = "2023.3.24" %}
+{% set name = "dask_ml" %}
+{% set version = "2024.4.4" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```

```suggestion
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name|replace("-","_") }}-{{ version }}.tar.gz
```
This will fix the `source/url`'s issue

---

Or just hardcode the name in the URL.

---

@JeanChristopheMorinPerso I was trying to avoid hardcoding the entire URL which lead to the slip up. Thanks @skupr-anaconda I will add your change :)

---

If you ask me, I see no harm in hardcoding the name in the URL.